articles/flask-database-integration-with-sqlalchemy/index.md

```diff
@@ -0,0 +1,329 @@
+        #store the form value
+        user_name = request.form["username"]
+        email = request.form["email"]
+        password = request.form["password"]
+
+        user = User(username = user_name, user_email = email, user_password = password)
+        
+        return username + " <br/> " + email
+    return render_template('register.html')
+```
+Next, we add the user and commit the changes to the database and we replace the return statement with “User registration successful”. By the end of the implementation, your registration route should like this.
+
+```python
+
+@app.route("/register", methods=["GET", "POST"])
+def register():
+    #check the request method to ensure the handling of POST request only
+    if request.method == "POST":
+        #store the form value
+        user_name = request.form["username"]
+        email = request.form["email"]
+        password = request.form["password"]
+
+        user = User(username = user_name, user_email = email, user_password = password)
+        db.session.add(user)
+        db.session.commit()
+        
+        return 'User registration successful'
+
+    return render_template('register.html')
+
+```
+
+## Test Running Our App
+Let’s test run our app. If you have the server running, terminate and restart. Now, you should have your server running, then proceed to the registration page on your web app. Let’s fill the data with this dummy data: username = Peter, email = peterdury@gmail.com, password= password.
+
+![Registration successful](/engineering-education/flask-database-integration-with-sqlalchemy/respinse.png)
+
+
+We received the registration successful message, but let’s check the database to check if our user data is stored.
+
+```bash
+>>> from app import db
+>>> from app.models import User
+>>> User.query.all()
+[<User 1>, <User 2>]
+```
+When you run the following command in sequential order, you should get a response indicating the new User added. In our case, we add a user when we added one manually, and due to the registration, we now have a new user.
+
+Let’s delve deeper and run a new command to query the User table by the username of the registered user.
+
+```bash
+>>>> user_data = User.query.all()
+>>>> for user in user_data:
+….    print(user.username)
+Demo user
+Peter
+```
+If you run the statement in sequential order, you should get the username data of all registered users in the database.
+
+## Conclusion
+This article has covered the usage of SQLAlchemy in building Flask Web applications. We also covered the types of database relationships that could exist in building database models. I recommend you look up how to implement a Many-to-Many database relationship.
+
+Try creating a contact form in your flask web app, then store the content in your custom database as a project to improve your skills when working with Flask-SQLAlchemy, and you can find the codebase for this article [here](https://github.com/corpsgeek/flask-sqlalchemy)
+
+
+
```

Comment:
Also, add a `Further reading`/`References` section containing the sources that you have used or any articles which could be read for better understanding.